src/rust/bitbox02-rust/src/keystore.rs

```diff
@@ -232,6 +230,48 @@ pub fn encrypt_and_store_seed(
     retain_seed(hal.random(), seed)
 }
 
+/// Restores a seed. This also unlocks the keystore with this seed.
+/// `password` is the password with which we encrypt the seed.
+pub fn encrypt_and_store_seed(
+    hal: &mut impl crate::hal::Hal,
+    seed: &[u8],
+    password: &str,
+) -> Result<(), Error> {
+    if bitbox02::memory::is_initialized() {
+        return Err(Error::Memory);
+    }
+    // We are in setup phase, so we clear memory first to guarantee clean RAM
+    encrypt_and_store_seed_internal(hal, seed, password)
+}
+
+/// Re-encrypts the seed with a (new) password
+pub fn re_encrypt_seed(
+    hal: &mut impl crate::hal::Hal,
+    seed: &[u8],
+    new_password: &str,
+) -> Result<(), Error> {
+    if !bitbox02::memory::is_initialized() {
+        return Err(Error::Unseeded);
+    }
```

Comment:
Fyi, seeded means an encrypted seed is stored, and initialized means it is seeded and a backup has been created.

In this case, `is_seeded()` is enough, because then there is an encrypted seed whose password we can change. And the `Unseeded` error will be correct too then :smile: